src/main/java/org/opentripplanner/transit/raptor/rangeraptor/standard/stoparrivals/Stops.java

```diff
@@ -134,6 +134,9 @@ private int round() {
         if(state.arrivedByTransfer()) {
             stopIndex = state.transferFromStop();
             state = stops[prevRound][stopIndex];
+            if(!state.arrivedByTransit()) {
+                return null;
+            }
```

Comment:
If you swap the `if`s (line 132 and 134) above this can be simplified:
```
        // We check for transfer before access, since a FLEX arrive on-board 
        // can be followed by a transfer
        if(state.arrivedByTransfer()) {
            stopIndex = state.transferFromStop();
            state = stops[prevRound][stopIndex];
        }
        if(state.arrivedByAccess()) { return null; }
```

Reply:
This makes sense